stdsimd-verify/tests/x86-intel.rs

```diff
@@ -0,0 +1,299 @@
+static U8x32: Type = Type::Unsigned(8, 32);
+static U8x8: Type = Type::Unsigned(8, 8);
+
+#[derive(Debug)]
+enum Type {
+    Float(u8, u8),
+    PrimFloat(u8),
+    PrimSigned(u8),
+    PrimUnsigned(u8),
+    Ptr(&'static Type),
+    Signed(u8, u8),
+    Unsigned(u8, u8),
+    Bool,
+}
+
+x86_functions!(static FUNCTIONS);
+
+#[derive(Deserialize)]
+struct Data {
+    #[serde(rename = "intrinsic", default)]
+    intrinsics: Vec<Intrinsic>,
+}
+
+#[derive(Deserialize)]
+struct Intrinsic {
+    rettype: String,
+    name: String,
+    tech: String,
+    #[serde(rename = "CPUID", default)]
+    cpuid: Vec<String>,
+    #[serde(rename = "parameter", default)]
+    parameters: Vec<Parameter>,
+    #[serde(default)]
+    instruction: Vec<Instruction>,
+}
+
+#[derive(Deserialize)]
+struct Parameter {
+    #[serde(rename = "type")]
+    type_: String,
+}
+
+#[derive(Deserialize)]
+struct Instruction {
+    name: String,
+}
+
+#[test]
+fn verify_all_signatures() {
+    let xml = include_bytes!("../x86-intel.xml");
+    let xml = &xml[..];
+    let data: Data = serde_xml_rs::deserialize(xml).expect("failed to deserialize xml");
+    let mut map = HashMap::new();
+    for intrinsic in data.intrinsics.iter() {
+        // This intrinsic has multiple definitions in the XML, so just ignore it.
+        if intrinsic.name == "_mm_prefetch" {
+            continue
+        }
+
+        // These'll need to get added eventually, but right now they have some
+        // duplicate names in the XML which we're not dealing with yet
+        if intrinsic.tech == "AVX-512" {
+            continue
+        }
+
+        assert!(map.insert(&intrinsic.name[..], intrinsic).is_none());
+    }
+
+    for rust in FUNCTIONS {
+        // This was ignored above, we ignore it here as well.
+        if rust.name == "_mm_prefetch" {
+            continue
+        }
+
+        // these are all AMD-specific intrinsics
+        if rust.target_feature.contains("sse4a") ||
+            rust.target_feature.contains("tbm") {
+            continue
+        }
+
+        let intel = match map.get(rust.name) {
+            Some(i) => i,
+            None => panic!("missing intel definition for {}", rust.name),
+        };
+
+        // Verify that all `#[target_feature]` annotations are correct, ensuring
+        // that we've actually enabled the right instruction set for this
+        // intrinsic.
+        assert!(intel.cpuid.len() > 0, "missing cpuid for {}", rust.name);
+        for cpuid in intel.cpuid.iter() {
+            // this is needed by _xsave and probably some related intrinsics,
+            // but let's just skip it for now.
+            if *cpuid == "XSS" {
+                continue
+            }
+
+            let cpuid = cpuid
+                .chars()
+                .flat_map(|c| c.to_lowercase())
+                .collect::<String>();
+
+            // Normalize `bmi1` to `bmi` as apparently that's what we're calling
+            // it.
```

Comment:
We might want to change the name of the target feature here to match intel's name before stabilization.